test/clj/rems/api/test_end_to_end.clj

```diff
@@ -0,0 +1,191 @@
+                       api-key applicant-id)))
+
+          (testing "submit application"
+            (assert-success
+             (api-call :post "/api/applications/submit" {:application-id application-id}
+                       api-key applicant-id)))
+
+          ;; we could start the pollers normally and wait for them to process the events here for better coverage
+          (email-poller/run)
+          (entitlements-poller/run)
+
+          (testing "email for new application"
+            (let [mail (last @email-atom)]
+              (is (= "handler@example.com" (:to mail)))
+              (is (.contains (:subject mail) "new application"))
+              (is (.startsWith (:body mail) "Dear E2E Handler,")))
+            (reset! email-atom []))
+
+          (testing "no entitlement yet"
+            (let [entitlements (api-call :get "/api/entitlements" nil
+                                         api-key owner-id)
+                  applications (set (map :application-id entitlements))]
+              (is (not (contains? applications application-id)))))
+
+          (testing "fetch application as handler"
+            (let [applications (api-call :get "/api/applications/todo" nil
+                                         api-key handler-id)
+                  todos (set (map (juxt :application/id :application/todo) applications))]
+              (is (contains? todos [application-id "new-application"])))
+            (let [application (api-call :get (str "/api/applications/" application-id) nil
+                                        api-key handler-id)]
+              (is (= "e2e test contents" (get-in application [:application/form :form/fields 0 :field/value])))
+              (is (= [license-id] (get-in application [:application/accepted-licenses (keyword applicant-id)]))
+                  application)))
+
+          (testing "approve application"
+            (assert-success
+             (api-call :post "/api/applications/approve" {:application-id application-id
+                                                          :comment "e2e approved"}
+                       api-key handler-id)))
+
+          (email-poller/run)
+          (entitlements-poller/run)
+
+          (testing "email for approved application"
+            (let [mail (last @email-atom)]
+              (is (= "applicant@example.com" (:to mail)))
+              (is (.contains (:subject mail) "approved"))
+              (is (.startsWith (:body mail) "Dear E2E Applicant,")))
+            (reset! email-atom []))
+
+          (testing "entitlement granted"
+            (let [[entitlement & others] (api-call :get (str "/api/entitlements?user=" applicant-id) nil
+                                                   api-key owner-id)]
+              (is (empty? others))
+              (is (= resource-ext-id (:resource entitlement)))
+              (is (not (:end entitlement)))))
```

Comment:
should we test the entitlement POST here too?

Reply:
You mean when the entitlements are posted to another system? Sounds like a good thing to test all integrations.

Reply:
I'll add it

Reply:
A fake email server would also be more end-to-end than mocking the email API.

Reply:
yeah, you're right, I'll add a TODO about that